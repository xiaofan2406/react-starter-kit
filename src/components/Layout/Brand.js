import React, { memo } from 'react';
import { APP_TITLE } from '../../utils/constants';
import logo from '../../assets/logo.svg';

const Brand = () => (
  <div className="flex items-center p-2">
    <img src={logo} alt="logo" className="spin h-10" />
    <span className="fade-in text-lg">{APP_TITLE}</span>
  </div>
);

export default memo(Brand);
