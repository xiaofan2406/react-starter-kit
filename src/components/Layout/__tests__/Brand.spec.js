/* @flow */
import React from 'react';
import { shallow } from 'enzyme';
import { Component } from '../Brand';

describe('Brand component', () => {
  afterEach(() => {
    jest.restoreAllMocks();
  });

  it('matches snapshot', () => {
    const props = { classes: {} };
    const component = shallow(<Component {...props} />);

    expect(component).toMatchSnapshot();
  });

  it('does not update on props changes', () => {
    const renderSpy = jest.spyOn(Component.prototype, 'render');
    const props = { classes: {} };
    const component = shallow(<Component {...props} />);

    expect(renderSpy).toHaveBeenCalledTimes(1);

    component.setProps({
      classes: { someClass: {} },
    });

    expect(renderSpy).toHaveBeenCalledTimes(1);
  });
});
