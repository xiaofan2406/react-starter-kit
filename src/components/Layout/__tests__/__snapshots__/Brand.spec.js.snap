// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Brand component matches snapshot 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 12px;
}

.emotion-0 > .logo {
  -webkit-animation: spin infinite 10s linear;
  animation: spin infinite 10s linear;
  height: 36px;
}

.emotion-0 > .title {
  -webkit-animation: fadeIn 2s ease;
  animation: fadeIn 2s ease;
  font-size: 18px;
}

<div
  className="emotion-0"
>
  <img
    alt="logo"
    className="logo"
    src="logo.svg"
  />
  <span
    className="title"
  >
    React Starter Kit
  </span>
</div>
`;
